src/VisualStudio/Core/Def/LanguageService/AbstractPackage.cs

```diff
@@ -27,16 +27,20 @@ internal IComponentModel ComponentModel
 
     protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
     {
-        await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
+        // Should only be called from a threadpool. Opinionated, as package load sequence thread switches are impactful.
+        Contract.ThrowIfTrue(JoinableTaskFactory.Context.IsOnMainThread);
 
-        await JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+        await base.InitializeAsync(cancellationToken, progress).ConfigureAwait(true);
```

Comment:
You can remove this its just noise, I've documented this on the base class and removed this across the entire VS tree. You don't need to call base.

Reply:
Sure, I can avoid doing the calls into AsyncPackage.InitializeAsync and AsyncPackage.OnAfterPackageLoadAsnc

Reply:
Done!